clang-tools-extra/docs/ReleaseNotes.rst

```diff
@@ -151,6 +151,10 @@ Changes in existing checks
   <clang-tidy/checks/bugprone/inc-dec-in-conditions>` check to ignore code
   within unevaluated contexts, such as ``decltype``.
 
+- Improved :doc:`bugprone-lambda-function-name
+  <clang-tidy/checks/bugprone/lambda-function-name>` check by ignoring
+  ``__func__`` macro in lambda captures and nested function declaration.
```

Comment:
I think the default parameter part should be included as well (+nits):
`check by ignoring the ``__func__`` macro in lambda captures, initializers of default parameters and nested function declarations.`